tests/functional/fx_firstrun_v1_sign_in.js

```diff
@@ -5,113 +5,99 @@
 define([
   'intern',
   'intern!object',
-  'intern/node_modules/dojo/node!xmlhttprequest',
-  'app/bower_components/fxa-js-client/fxa-client',
   'tests/lib/helpers',
   'tests/functional/lib/helpers'
-], function (intern, registerSuite, nodeXMLHttpRequest, FxaClient,
-  TestHelpers, FunctionalHelpers) {
+], function (intern, registerSuite, TestHelpers, FunctionalHelpers) {
   var config = intern.config;
-  var AUTH_SERVER_ROOT = config.fxaAuthRoot;
   var PAGE_URL = config.fxaContentRoot + 'signin?context=iframe&service=sync';
   var NO_REDIRECT_URL = PAGE_URL + '&haltAfterSignIn=true';
 
   var email;
   var PASSWORD = '12345678';
-  var client;
 
-  var listenForFxaCommands = FunctionalHelpers.listenForWebChannelMessage;
+  var thenify = FunctionalHelpers.thenify;
+
+  var clearBrowserState = thenify(FunctionalHelpers.clearBrowserState);
+  var createUser = FunctionalHelpers.createUser;
+  var fillOutSignIn = thenify(FunctionalHelpers.fillOutSignIn);
+  var noSuchElement = FunctionalHelpers.noSuchElement;
+  var openPage = thenify(FunctionalHelpers.openPage);
   var respondToWebChannelMessage = FunctionalHelpers.respondToWebChannelMessage;
-  var testIsBrowserNotified = function (context, message) {
+  var testElementExists = FunctionalHelpers.testElementExists;
+
+  var testIsBrowserNotified = function (message) {
     message = message.replace(/:/g, '-');
     return function () {
-      return context.remote
+      return this.parent
        .findByCssSelector('#message-' + message)
        .end();
     };
   };
 
   registerSuite({
-    name: 'Firstrun sign_in',
+    name: 'Firstrun v1 sign_in',
 
     beforeEach: function () {
-      var self = this;
       email = TestHelpers.createEmail();
-      client = new FxaClient(AUTH_SERVER_ROOT, {
-        xhr: nodeXMLHttpRequest.XMLHttpRequest
-      });
-
-      return client.signUp(email, PASSWORD, { preVerified: true })
-        .then(function () {
-          return FunctionalHelpers.clearBrowserState(self);
-        });
-    },
 
-    afterEach: function () {
-      return FunctionalHelpers.clearBrowserState(this);
+      return this.remote
+        .then(clearBrowserState(this));
     },
 
-    'sign in with an already existing account': function () {
-      var self = this;
-
-      return FunctionalHelpers.openPage(this, PAGE_URL, '#fxa-signin-header')
-        .execute(listenForFxaCommands)
-
-        .then(respondToWebChannelMessage(self, 'fxaccounts:can_link_account', { ok: true } ))
-
-
-        .then(function () {
-          return FunctionalHelpers.fillOutSignIn(self, email, PASSWORD);
-        })
+    'verified': function () {
+      return this.remote
+        .then(createUser(email, PASSWORD, { preVerified: true }))
```

Comment:
As @vbudhram notes elsewhere, this file also has a lot of commonality that looks like it could be abstracted into a `setupTest` function.